easybuild/tools/module_generator.py

```diff
@@ -197,6 +198,51 @@ def prepend_paths(self, key, paths, allow_abs=False, expand_relpaths=True):
         """
         return self.update_paths(key, paths, prepend=True, allow_abs=allow_abs, expand_relpaths=expand_relpaths)
 
+    def modulerc(self, module_version=None):
+        """
+        Generate contents of .modulerc file, in Tcl syntax (compatible with all module tools, incl. Lmod)
+
+        :param module_version: specs for module-version statement (dict with 'modname', 'sym_version' & 'version' keys)
+        """
+        modulerc = [ModuleGeneratorTcl.MODULE_SHEBANG]
+
+        if module_version:
+            if isinstance(module_version, dict):
+                expected_keys = ['modname', 'sym_version', 'version']
+                if sorted(module_version.keys()) == expected_keys:
+
+                    module_version_statement = "module-version %(modname)s %(sym_version)s"
+
+                    # for Environment Modules we need to guard the module-version statement,
+                    # to avoid "Duplicate version symbol" warning messages where EasyBuild trips over,
+                    # which occur because the .modulerc is parsed twice
+                    # cfr. https://sourceforge.net/p/modules/mailman/message/33399425/
+                    if modules_tool().__class__ == EnvironmentModulesC:
+
+                        modname, sym_version, version = [module_version[key] for key in expected_keys]
+
+                        # determine module name with symbolic version
+                        if version in modname:
+                            # take a copy so we don't modify original value
+                            module_version = copy.copy(module_version)
+                            module_version['sym_modname'] = modname.replace(version, sym_version)
+                        else:
+                            raise EasyBuildError("Version '%s' does not appear in module name '%s'", version, modname)
+
+                        module_version_statement = '\n'.join([
+                            'if {"%(sym_modname)s" eq [module-info version %(sym_modname)s]} {',
```

Comment:
I know the explanation is referenced above  in the URL but this behaviour is very non-intuitive, a comment could really help to understand this in 6 months: 
```
If you pass a string version to "module-info version" for a version that doesn't exist you get the same string back as
output. However after you've used module-version to define that as a "valid" version you get back the real modulefile name
```